Lab3/src/L3Lab.scala

```diff
@@ -0,0 +1,133 @@
+import scala.annotation.tailrec
+
+object L3Lab extends App {
+  // Zadanie 1
+  def doubleListSplitBy(list: List[Double], predicate1: Double => Boolean, predicate2: Double => Boolean): (List[Double], List[Double]) = {
+    def listFilter(list: List[Double], predicate: Double => Boolean): List[Double] = list match {
+      case Nil => Nil
+      case head :: tail => if (predicate(head)) head :: listFilter(tail, predicate) else listFilter(tail, predicate)
+    }
+
+    (listFilter(list, predicate1), listFilter(list, predicate2))
```

Comment:
ogólnie fajnie wygląda struktura i funkcja bardzo reużywalna, ale niestety w tym przypadku niewydajna jak cholera. Mieliśmy listy ujemne i ujemne nieparzyste. Zamiast załatwić to przechodząc przez listę raz, przechodzisz dwa. Dla dużych / głębokich struktur to może być kosztowne.